server/controllers/GroupCtr.js

```diff
@@ -0,0 +1,133 @@
+
+import db from '../models';
+
+
+const GroupCtrl = {
+
+  /**
+   *  - Create a user
+   * @param {Object} req Request Object
+   * @param {Object} res Response Object
+   * @returns {void} Returns void
+   */
+  createGroup(req, res) {
+    db.Groups.findOrCreate({
+      where: { name: req.body.name },
+      defaults: {
+        description: req.body.description,
+        creator: req.decoded.username
+      }
+    }).spread((group, created) => {
+      if (created) {
+        group.addUser(req.decoded.userId);
+        return res.status(201).send({
+          id: group.id,
+          name: group.name,
+          description: group.description,
+          creator: group.creator
+        });
+      }
+      return res.status(409).send({ message: 'Group already exist' });
+    });
+  },
+  /** Retrieve all Group of a User
+   * @param {Object} req Request Object
+   * @param {Object} res Response Object
+   * @returns {object} Returns all user Groups
+   */
+
+  retrieveAllGroup(req, res) {
+    db.Users.findOne({
+      where: {
+        username: req.decoded.username
+      }
+    }).then((user) => {
+      user.getGroups().then((groups) => {
+        if (groups.length === 0) {
+          return res.status(200).send({
+            message: 'You do not belong to any group'
+          });
+        }
+        return res.status(200).send(groups);
+      });
+    }).catch(error => res.status(500).send({
+      message: 'server error'
+    }));
+  },
+  /** Add User To Group
+   * @param {Object} req Request Object
+   * @param {Object} res Response Object
+   * @returns {object} Returns User
+   */
+
+  addUserToGroup(req, res) {
+    const groupId = req.params.id;
```

Comment:
You should validate `id` to ensure it's a number. Hence sequelize will return error